File: Array/removeElement.py

class Solution:
    #def removeElement(self, nums: List[int], val: int) -> int:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        while True:
            try:
                nums.remove(val)
            except:
                break
        return len(nums)

    def removeElement1(self, nums, val):
        i = 0
        for j in nums:
            if j != val:
                nums[i] = j
                i+=1
        return i

File: Array/removeElement.cpp
# include <vector>
using std::vector;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i = 0;
        for (int j = 0; j < nums.size(); j++) {
            if (nums[j] != val) {
                nums[i] = nums[j];
                i++;
            }
        }
        
        return i;
    }
    int removeElement1(vector<int>& nums, int val) {
        int n = nums.size();
        auto it = nums.begin();
        
        int removeCount = 0;
        while (it!=nums.end()) {
            if (*it==val) {
                nums.erase(it);
                removeCount++;
            } else {
                it++;
            }
        }
        
        return n-removeCount;
    }  
};